developer/src/kmc-copy/src/KeymanProjectCopier.ts

```diff
@@ -0,0 +1,561 @@
+        result.artifacts[fullPath] = {
+          data: this.callbacks.loadFile(fullPath),
+          filename: this.generateNewFilename(fullPath, outputPath)
+        }
+      }
+    }
+  }
+
+  private copyProjectFile(project: KeymanDeveloperProject, filename: string, outputPath: string, source: string, result: CopierResult): boolean {
+    for(const file of project.files) {
+      const copier = this.copiers[<KeymanFileTypes.Source> file.fileType] ?? this.copyGenericFile.bind(this);
+
+      const subOutputPath = this.normalizePath(this.calculateNewFilePath(project.projectPath, file.filePath));
+      const subFilename = this.normalizePath(project.resolveInputFilePath(file));
+
+      // Ignore errors because we will continue to do a best effort
+      copier(project, subFilename, subOutputPath, source, result);
+    }
+
+    if(project.options.version == "2.0") {
+      // For version 2.0 projects, we also copy every file in the folder
+      // except for project.buildpath and .kpj.user
+      this.copyFolder(project.projectPath, outputPath, [/\.kpj\.user/, project.resolveBuildPath()], result);
+    }
+
+    // Rewrite the project
+
+    project.options = new KeymanDeveloperProjectOptions('2.0');
+    const writer = new KPJFileWriter();
+    const data = new TextEncoder().encode(writer.write(project));
+
+    result.artifacts[filename] = {
+      data,
+      filename: this.generateNewFilename(filename, outputPath)
+    };
+
+    return true;
+  }
+
+  private copyGenericFile(project: KeymanDeveloperProject, filename: string, outputPath: string, source: string, result: CopierResult): boolean {
+    if(result.artifacts[filename]) {
+      return true;
+    }
+
+    const data = this.callbacks.loadFile(filename);
+    if(!data) {
+      // We'll add an null artifact and do a best effort, rather
+      // than fail the process altogether.
+      result.artifacts[filename] = {
+        data: null,
+        filename: this.generateNewFilename(filename, outputPath)
+      };
+
+      // Only warn that the file is missing if it isn't an output file,
+      // because a clean project won't contain any outputs (binaries)
+      if(KeymanFileTypes.binaryTypeFromFilename(filename) === null) {
+        this.callbacks.reportMessage(CopierMessages.Warn_FileNotFound({filename, newFilename: result.artifacts[filename].filename}));
+      }
+
+      return false;
+    }
+
+    result.artifacts[filename] = {
+      data,
+      filename: this.generateNewFilename(filename, outputPath)
+    };
+
+    return true;
+  }
+
+  private copySourceFile(project: KeymanDeveloperProject, filename: string, outputPath: string, source: string, result: CopierResult): boolean {
+    return this.copyGenericFile(project, filename, this.callbacks.path.join(this.outPath, 'source'), source, result);
+  }
+
+  private copyModelTsSourceFile(project: KeymanDeveloperProject, filename: string, outputPath: string, source: string, result: CopierResult): boolean {
```

Comment:
So this also copies tsv wordlists?

Reply:
Yes, just like `copyKmnSourceFile` copies icons, etc. Each function is responsible for copying the files that are referenced by its input file.